app.go

```diff
@@ -449,22 +456,46 @@ func ValidateApp(opts ...Option) error {
 	return app.Err()
 }
 
+// logBuffer will buffer all messages until a logger has been
+// initialized.
+type logBuffer struct {
+	entries []fxlog.Entry
+	logger  fxlog.Logger
+}
+
+// Log buffers or logs an entry.
+func (l *logBuffer) Log(entry fxlog.Entry) {
+	if l.logger == nil {
+		l.entries = append(l.entries, entry)
+	} else {
+		l.logger.Log(entry)
+	}
+}
+
+// Connect flushes out all buffered entries to a logger and resets them.
+func (l *logBuffer) Connect(logger fxlog.Logger) {
+	l.logger = logger
+	for _, e := range l.entries {
+		logger.Log(e)
```

Comment:
Use l.logger.Log(e) since it's already set.

Reply:
Is this for readability? I'm curious to learn about which way is better and why.